File: scala-3/euler/src/main/scala/com/rbleggi/euler/P09.scala
package com.rbleggi.euler

@main def mainP09(): Unit = {
  for (a <- 1 to 998) {
    for (b <- a + 1 to 999) {
      val c = 1000 - a - b
      if (a * a + b * b == c * c) {
        println(s"The Pythagorean triplet is: a=$a, b=$b, c=$c")
        println(s"The product abc is: ${a * b * c}")
        return
      }
    }
  }
}

File: kotlin/euler/src/main/kotlin/com/rbleggi/euler/P09.kt
package com.rbleggi.euler

fun mainP09() {
    for (a in 1..998) {
        for (b in a + 1..999) {
            val c = 1000 - a - b
            if (a * a + b * b == c * c) {
                println("The Pythagorean triplet is: a=$a, b=$b, c=$c")
                println("The product abc is: ${a * b * c}")
                return
            }
        }
    }
}
